### PR TITLE
feat(search-input): add component

### PR DESCRIPTION
**Pull request by colbr**
## 1. Pull request checklist

- [x] PR name follows [conventional commit spec](https://www.conventionalcommits.org/en/v1.0.0/)

## 2. What is the old behaviour? (if applicable)

`search` `Input` didn't match Figma.

## 3. What is the new behaviour?

`search` `Input` now matches Figma.

## 4. How can this behaviour be tested? (if applicable)

1. Open Storybook
2. Navigate to `Inputs > Search`
3. Check that the component matches [Figma](https://www.figma.com/file/wTOVNZonl9MmEzORwBHDuq/ZERO-Design-System-(zDS)?type=design&node-id=4546%3A36921&t=TObYKB5R5LIbt8S5-1)

### CODE REVIEW

**Review by domw30 — APPROVED** [review]
Nice 👍

File: src/components/Input/SearchInput.module.scss [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+@mixin path($color, $styles) {
+    svg {
+        path {
+            stroke: $color;
+        }
+    }
+}
+
+.SearchWrapper {
+    border-radius: 1000px !important; // make it pill shaped
```
Comment (ratik21) [inline src/components/Input/SearchInput.module.scss:12]: what does `!important` do?

File: src/components/Input/SearchInput.module.scss [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+}
+
+.SearchIcon {
+    &[data-is-searching] {
```
Comment (ratik21) [inline src/components/Input/SearchInput.module.scss:16]: so here `data-is-searching` is a boolean value, and the css is picked on the basis of it being `true`/`false`?

**Review by ratik21 — APPROVED** [review]
had 1-2 css doubts

File: src/components/Input/SearchInput.module.scss (hunk 1 above)
Comment (colbr) [reply src/components/Input/SearchInput.module.scss:12]: It overrides previous set values for `border-radius` - for some reason this `border-radius` is being overridden. Will have to work out why it's getting overridden in another PR, but it's not critical right now.

File: src/components/Input/SearchInput.module.scss (hunk 2 above)
Comment (colbr) [reply src/components/Input/SearchInput.module.scss:16]: Yup! It's basically saying `if data-is-searching has a value`, and in the component we set `data-is-searching={isSearching ? '' : null}` ☺️

Comment (ratik21) [reply src/components/Input/SearchInput.module.scss:16]: got it, thanks

**Pull request merged**